extractor/gomodextractor.go

```diff
@@ -100,27 +100,74 @@ type GoModExprCommentWrapper struct {
 	expr modfile.Expr
 }
 
+func minInt(a int, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func maxInt(a int, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func lexMin(a1 int, a2 int, b1 int, b2 int) (int, int) {
+	if a1 < b1 {
+		return a1, a2
+	} else if a1 > b1 {
+		return b1, b2
+	} else {
+		return a1, minInt(b1, b2)
+	}
+}
+
+func lexMax(a1 int, a2 int, b1 int, b2 int) (int, int) {
+	if a1 < b1 {
+		return b1, b2
+	} else if a1 > b1 {
+		return a1, a2
+	} else {
+		return a1, maxInt(b1, b2)
+	}
+}
+
 func extractGoModComments(tw *trap.Writer, expr modfile.Expr, exprlbl trap.Label) {
+	comments := expr.Comment()
+
+	if len(comments.Before) == 0 && len(comments.Suffix) == 0 && len(comments.After) == 0 {
+		return
+	}
+
 	// extract a pseudo `@commentgroup` for each expr that contains their associated comments
 	grouplbl := tw.Labeler.LocalID(GoModExprCommentWrapper{expr})
 	dbscheme.CommentGroupsTable.Emit(tw, grouplbl)
 	dbscheme.DocCommentsTable.Emit(tw, exprlbl, grouplbl)
 
-	comments := expr.Comment()
+	var allComments []modfile.Comment
+	allComments = append(allComments, comments.Before...)
+	allComments = append(allComments, comments.Suffix...)
+	allComments = append(allComments, comments.After...)
 
+	var startLine, startCol, endLine, endCol int = 0, 0, 0, 0
+	var first bool = true
 	idx := 0
-	for _, comment := range comments.Before {
-		extractGoModComment(tw, comment, grouplbl, idx)
-		idx++
-	}
-	for _, comment := range comments.Suffix {
-		extractGoModComment(tw, comment, grouplbl, idx)
-		idx++
-	}
-	for _, comment := range comments.After {
+	for _, comment := range allComments {
 		extractGoModComment(tw, comment, grouplbl, idx)
 		idx++
+		commentEndCol := comment.Start.LineRune + len(comment.Token)
```

Comment:
Our end positions are generally inclusive, so I think this should be

```suggestion
		commentEndCol := comment.Start.LineRune + len(comment.Token) - 1
```

Reply:
Also, do I interpret the test failures on Windows correctly to mean that `comment.Token` includes a `\r` if the line ending is `\r\n`? If so, we'll probably want to chop it off.